learning/tour-of-beam/frontend/lib/pages/tour/state.dart

```diff
@@ -136,11 +142,123 @@ class TourNotifier extends ChangeNotifier with PageStateMixin<void> {
     if (content == null) {
       return;
     }
-    final taskSnippetId = content.taskSnippetId;
-    await _setPlaygroundSnippet(taskSnippetId);
-    _isShowingSolution = false;
+
+    await _updateHasSavedSnippet();
+    _setSnippetTypeSavedIfCan();
+    await _setCurrentSnippetFromType();
+    notifyListeners();
+  }
+
+  // Save user code.
```

Comment:
Save user code part will be extracted into a separate class in the [Save user code for anonymous user](https://github.com/apache/beam/issues/25283) task.